src/ai/defaultai.cc

```diff
@@ -3330,26 +3338,239 @@ void DefaultAI::check_flag_distances(const Time& gametime) {
+						                  "diploscore %d%s\n",
+						                  static_cast<unsigned int>(mypn), myts_s.c_str(),
+						                  planned_action == Widelands::DiplomacyAction::kAcceptInvite ?
+                                       "invitation" :
+                                       "join request",
+						                  static_cast<unsigned int>(planned_opn), planned_other_score,
+						                  planned_log_append_text.c_str());
+						game().send_player_diplomacy(
+						   mypn,
+						   (planned_action == Widelands::DiplomacyAction::kAcceptInvite ?
+                         Widelands::DiplomacyAction::kRefuseInvite :
+                         Widelands::DiplomacyAction::kRefuseJoin),
+						   planned_opn);
+					}
+				}
+				planned_action = pda.action == Widelands::DiplomacyAction::kInvite ?
+                                Widelands::DiplomacyAction::kAcceptInvite :
+                                Widelands::DiplomacyAction::kAcceptJoin;
+				planned_opn = pda.sender;
+				planned_other_score = diploscore;
+				plan_priority = priority;
+				if (g_verbose) {
+					planned_log_append_text = other_team_score_str;
+					/* verb_ */ log_dbg_time(
+					   gametime,
+					   "AI Diplomacy: Player(%d) plans to accept the %s of player (%d) with priority %d",
+					   static_cast<unsigned int>(mypn),
+					   pda.action == Widelands::DiplomacyAction::kInvite ? "invitation" : "join request",
+					   pda.sender, plan_priority);
+				}
+			}
+		}
+	}
+
+	// Execute only one action that changes team line-ups
+	if (planned_action != kNoAction) {
+		assert(planned_action == Widelands::DiplomacyAction::kLeaveTeam ||
+		       planned_action == Widelands::DiplomacyAction::kAcceptInvite ||
+		       planned_action == Widelands::DiplomacyAction::kAcceptJoin);
+		if (g_verbose) {
+			std::string action_str = "join request";
+			switch (planned_action) {
+			case Widelands::DiplomacyAction::kLeaveTeam:
+				verb_log_dbg_time(gametime, "AI Diplomacy: Player(%d) leaves team (%d)%s.\n",
+				                  static_cast<unsigned int>(mypn), static_cast<unsigned int>(mytn),
+				                  planned_log_append_text.c_str());
+				break;
+
+			case Widelands::DiplomacyAction::kAcceptInvite:
+				action_str = "invitation";
+				FALLS_THROUGH;
+			case Widelands::DiplomacyAction::kAcceptJoin:
+				assert(planned_other_score != kNoScore);
+				verb_log_dbg_time(
+				   gametime,
+				   "AI Diplomacy: Player(%d)%s accepts the %s of player (%d) with diploscore %d%s.\n",
+				   static_cast<unsigned int>(mypn), myts_s.c_str(), action_str.c_str(),
+				   static_cast<unsigned int>(planned_opn), planned_other_score,
+				   planned_log_append_text.c_str());
+				break;
+
+			default:
+				NEVER_HERE();
+			}
+		}
+		game().send_player_diplomacy(mypn, planned_action, planned_opn);
+		player_statistics.recalculate_team_power();
+
+		// Team change invalidates statistics
+		return;
```

Comment:
`recalculate_team_power()` is useless here, it's effectively a no-op that just wastes some cycles on iterating over the old statistics again.

Reply:
oops you are right will undo this